src/Entity/CivicrmEntity.php

```diff
@@ -21,6 +22,27 @@
  */
 class CivicrmEntity extends ContentEntityBase {
 
+  /**
+   * {@inheritdoc}
+   */
+  public function save() {
+    // Set ::drupal_crud to indicate save is coming from Drupal.
+    $this->drupal_crud = TRUE;
+    $result = parent::save();
+    $this->drupal_crud = FALSE;
```

Comment:
I think this should probably have a `try { ... } finally { ... }` to make sure that `$this->drupal_crud = FALSE;` is run even if an exception is thrown while saving